packages/botonic-api/src/rest/routes/validation/events.ts

```diff
@@ -0,0 +1,263 @@
+import { MessageEventFrom } from '@botonic/core/lib/models/events/message'
+import { EventTypes } from '@botonic/core/src/models/events'
+import { ConnectionEventStatuses } from '@botonic/core/src/models/events/connections/index'
+import {
+  MessageEventAck,
+  MessageEventTypes,
+} from '@botonic/core/src/models/events/message'
+import { Request } from 'express'
+import {
+  body,
+  checkSchema,
+  ParamSchema,
+  Result,
+  Schema,
+  validationResult,
+} from 'express-validator'
+import { ValidatorsSchema } from 'express-validator/src/middlewares/schema'
+
+const inQuery: ParamSchema = { in: ['query'] }
+const inParams: ParamSchema = { in: ['params'] }
+const inBody: ParamSchema = { in: ['body'] }
+const isRequired: ValidatorsSchema = { notEmpty: true }
+const isOptional: ParamSchema = { optional: { options: { nullable: true } } }
+const isNaturalNumber: ParamSchema = {
+  isNumeric: { options: { no_symbols: true } },
+}
+const isNumeric: ParamSchema = { isNumeric: true }
+const isBoolean: ParamSchema = { isBoolean: true }
+const isDateTime: ParamSchema = {
+  isISO8601: { options: { strict: true, strictSeparator: true } },
+}
+const isIn = (valueList: string[]): ParamSchema => {
+  return {
+    isIn: { options: [valueList] },
+    errorMessage: `Invalid value. Allowed values: ${valueList}`,
+  }
+}
+const equals = (text: string): ParamSchema => {
+  return {
+    equals: { options: [text] },
+    errorMessage: `Invalid value. Value should be '${text}'`,
+  }
+}
+const toInt: ParamSchema = { toInt: true }
+
+export const limitParamSchema: ParamSchema = {
+  ...inQuery,
+  ...isOptional,
+  ...isNaturalNumber,
+  ...toInt,
+}
+export const offsetParamSchema = limitParamSchema
+
+export const eventIdParamSchema: ParamSchema = { ...inParams, ...isRequired }
+
+export const buttonsSchema: Schema = {
+  buttons: {
+    ...inBody,
+    ...isOptional,
+    isArray: true,
+    errorMessage: 'buttons have to be an array',
+  },
+  'buttons.*.title': { ...inBody, ...isRequired },
```

Comment:
Question, does `*` have a special effect here? Or you just want the key to have that value? It looks like this is the value to me.

Reply:
Yes, it's how express validator tackles the validation of arrays -> https://express-validator.github.io/docs/wildcards.html